src/site/topic14-careful.rst

```diff
@@ -0,0 +1,211 @@
+
+    public void pumpItUp(Rectangle rect) {
+        rect.setWidth(2.0 * rect.getWidth());
+    }
+
+* ``pumpItUp`` is a method that will double the area of a rectangle
+
+* Think of what will happen when we run this
+
+.. code-block:: java
+    :linenos:
+
+    Square mySquare = new Square(10);
+    pumpItUp(mySquare);
+
+
+* Since our ``Square`` inherits from ```Rectangle``, we get ``setWidth``, but this will then cause the square instance to have a length and width not equal
+    * Thus, our ``Square`` is no longer a square
+
+* This can be fixed by overriding the ``setWidth`` (and ``setLength``) methods in the ``Square`` class
+
+.. code-block:: java
+    :linenos:
+
+        // Add to Square class to override
+        // Rectangle's setters
+        public void setWidth(double width) {
+            super.setWidth(width);
+            super.setLength(width);
+        }
+
+        public void setLength(double length) {
+            this.setWidth(length);
+        }
+
+* What happens now if we call this?
+
+.. code-block:: java
+    :linenos:
+
+    Square mySquare = new Square(10);
+    pumpItUp(mySquare);
+
+* This will then cause our square to not double in size, but quadruple, which is a problem
+* This means we can't substitute the ``Rectangle`` for a ``Square`` for ``pumpItUp``
+
+* However, this can be fixed by changing our ``pumpItUp`` method
+
+.. code-block:: java
+    :linenos:
+
+    public void pumpItUp(Rectangle rect) {
+        if (rect instanceof Square) {
+            rect.setWidth(Math.sqrt(2.0) * rect.getWidth());
+        } else {
+            rect.setWidth(2.0 * rect.getWidth());
+        }
+    }
+
+* Now this *solves* it
+* Except, `Hyrum's Law <https://www.hyrumslaw.com/>`_ says that all observable behaviours, intentional or not, will be depended on by somebody
+* So, someone out there depends on the fact that ``pumpItUp`` is quadrupling the ``Square``, even though it honestly shouldn't be
+
+* Ok, so, we can fix this by adding another method and changing ``pumpItBack`` for the person depending on the problematic functionality
+
+.. code-block:: java
+    :linenos:
+
+    public void pumpItUp(Rectangle rect) {
+        rect.setWidth(2.0 * rect.getWidth());
+    }
+
+    public void myPumpItUp(Rectangle rect) {
+        if (rect instanceof Square) {
+            rect.setWidth(Math.sqrt(2.0) * rect.getWidth());
+        } else {
+            pumpItUp(rect);
+        }
+    }
+
+* NOW this *solves* it
+* But, now we have a function saying: if it's a ``Square`` do one thing, if it's a ``Rectangle`` do another thing
+    * So... it would seem that for our needs here, a ``Square`` is **not** a ``Rectangle``
+* We also have two pieces of code trying to do the same thing
+    * What happens if ``Rectangle`` gets extended again? Write another version of the method?
+
+* We ended up doing a lot of extra work for no reason at all
+* Our code got more complex
+* It's going to be a lot easier to just not use inheritance here
```

Comment:
It's probably also worth mentioning a better option: [composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance). If you want to code share, a `Square` could _use_ a `Rectangle` and delegate the square specific logic to that.